### PR TITLE
Miscellaneous minor fixes and improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Mainly fix unit tests for default encoding on Windows plus some other minor fixes and improvements.

### CODE REVIEW

**Comment by sschuberth** [comment]
@svanoort Would you mind having a look?

File: src/main/java/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig.java [hunk 1]
```diff
@@ -30,7 +30,6 @@
  */
 public class BlockableBuildTriggerConfig extends BuildTriggerConfig {
     private final BlockingBehaviour block;
-    public boolean buildAllNodesWithLabel;
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig.java]: Breaks binary compatibility. If unused, should be marked as deprecated

File: src/main/resources/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig/config.jelly [hunk 2]
```diff
@@ -25,13 +25,7 @@ THE SOFTWARE.
...
     <f:textbox />
   </f:entry>
-  <!-- TODO: replace with <f:optionalProperty> -->
-  <f:optionalBlock field="block" title="${%Block until the triggered projects finish their builds}" checked="${instance.block!=null}">
-    <j:set var="descriptor" value="${app.getDescriptorOrDie(descriptor.getPropertyType(field).clazz)}" />
-    <j:set var="instance" value="${instance[field]}"/>
-    <st:include from="${descriptor}" page="${descriptor.configPage}" />
-  </f:optionalBlock>
-
+  <f:optionalProperty field="block" title="${%Block until the triggered projects finish their builds}" />
```
Comment (oleg-nenashev) [inline src/main/resources/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig/config.jelly]: Did you manually test it?

File: src/test/resources/hudson/plugins/parameterizedtrigger/test/BuildInfoExporterTest/testMigrateFrom221/jobs/upstream/config.xml [hunk 3]
```diff
@@ -104,11 +102,10 @@
               <color>RED</color>
             </failureThreshold>
           </block>
-          <buildAllNodesWithLabel>false</buildAllNodesWithLabel>
```
Comment (oleg-nenashev) [inline src/test/resources/hudson/plugins/parameterizedtrigger/test/BuildInfoExporterTest/testMigrateFrom221/jobs/upstream/config.xml]: In this test change you hide the problem you introduce by breaking the compatibility

**Comment by oleg-nenashev** [comment]
:-1: for merge in the current state

File: src/main/java/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig.java (hunk 1 above)
Comment (sschuberth) [inline src/main/java/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig.java]: I'll drop this change as I not not want to mix this PR of mine with code deprecation.

File: src/main/resources/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig/config.jelly (hunk 2 above)
Comment (sschuberth) [inline src/main/resources/hudson/plugins/parameterizedtrigger/BlockableBuildTriggerConfig/config.jelly]: Yes, it works according to my test.

File: src/test/resources/hudson/plugins/parameterizedtrigger/test/BuildInfoExporterTest/testMigrateFrom221/jobs/upstream/config.xml (hunk 3 above)
Comment (sschuberth) [inline src/test/resources/hudson/plugins/parameterizedtrigger/test/BuildInfoExporterTest/testMigrateFrom221/jobs/upstream/config.xml]: Dropping this change, too.

**Comment by sschuberth** [comment]
@oleg-nenashev I've updated the PR to address your comments.

**Comment by svanoort** [comment]
Apologies for the delay.  It looks like my filter that was supposed to tag direct github mentions (so they don't get buried in the 50+ per day) needs some tweaks, and this slipped through. 

:+1:  from me for this, and I'll merge tomorrow if no objections are raised

**Comment by sschuberth** [comment]
I've pushed an update that removes the multi-byte assertions for non-multi-byte default character sets, as I believe there's no point in comparing against "???"-like strings.

**Comment by oleg-nenashev** [comment]
:+1:

**Comment by sschuberth** [comment]
@svanoort I think it's ready now <wink> ;-)

**Comment by svanoort** [comment]
:+1:  and merge as a maintainer

**Pull request merged**